scripts/abi_artifact_generator/templates/setup_arguments.S.jinja_template

```diff
@@ -0,0 +1,53 @@
+/*#-
+This template file is distributed under the MIT License. See LICENSE.md for details.
+#*/
+
+/*(- generation_notice )*/
+
+/*{ if current_style == 'gcc' -}*/
+#include "../abi.h"
+
+#if defined(TARGET_x86_64) || defined(TARGET_i386)
+.intel_syntax noprefix
+#endif
+/*{- elif current_style == 'msvc' -}*/
+ifndef rax
+.model flat, c
+endif
+
+extern randomized_state: /*( intel_register_type )*/
+extern printable_data: /*( intel_register_type )*/
+extern saved_return_address: /*( intel_register_type )*/
+
+/*{ for name, arguments in argument_functions.items() -}*/
+extern test_/*( name )*/: proc
+/*{ endfor }*/
+
+.code
+/*{- endif }*/
+
+/*{ for name, arguments in argument_functions.items() }*/
```

Comment:
Yep, but the big problem is that we cannot really afford to use a register for this, so it needs to either be some weird trick, or self-modifying code.

Either way, comment is a good idea.

Reply:
> we cannot really afford to use a register for this

What do you mean? What would be wrong with my example above?

If we just add a call + ret, in x86-64 means we use (and free) a stack slot, in other architecture we clobber the link register, which we were going to clobber anyway a few lines below.

Reply:
Ah wait, right, I was thinking about it from the ABI perspective. I guess it can work. Sure, it's heavily architecture-specific, but we're already doing architecture-specific stuff here.

Reply:
Closing this for now.